src/script/conversation/ConversationRepository.ts

```diff
@@ -1626,12 +1626,11 @@ export class ConversationRepository {
    * @returns Resolves when user was removed from the conversation
    */
   private async leaveConversation(conversationEntity: Conversation, clearContent: boolean) {
-    if (clearContent) {
-      this.clearConversation(conversationEntity, false);
-    }
-
     const userQualifiedId = this.userState.self().qualifiedId;
-    return this.removeMemberFromConversation(conversationEntity, userQualifiedId);
+
+    return clearContent
+      ? this.clearConversation(conversationEntity, true)
+      : this.removeMemberFromConversation(conversationEntity, userQualifiedId);
```

Comment:
This is the main change - the `true` param in clearConversation method is `leaveConversation` - so when called with `true`, we clear the conversation locally and leave it. Previously it was called with false - conversation was not cleared and we were calling just `removeMemberFromConversation` - leaving a conversation without clearing the local store, even though user has marked the checkbox